datatypes/src/main/java/org/hyperledger/besu/datatypes/Hash.java

```diff
@@ -61,6 +61,16 @@ public static Hash hash(final Bytes value) {
     return new Hash(keccak256(value));
   }
 
+  /**
+   * Convert value to sha256 hash.
+   *
+   * @param value the bytes to hash
+   * @return the hash of those bytes
+   */
+  public static Hash sha256(final Bytes value) {
+    return new Hash(org.hyperledger.besu.crypto.Hash.sha256(value));
```

Comment:
I don't like this.  The datatype Hash object has always meant a container for a Keccak-256 hash, because that was the only hash type used internally.  We either need to add a type field showing what type the hash is, and check it when we use it (the critical part) or create a new HashSha256 type.

Reply:
The solution is probably to promote a datatypes variant of VersionedHash into the datatypes package.

Reply:
lol I had initially done that, but wasn't sure if the EVM wanted to be Cancun aware. Will promote it and update MessageFrame signatures to use it.

Reply:
Having the EVM Cancun aware is fine.  In fact it is preferable.  The message frame already has new fields for the versioned hash, adding a new type doesn't add any incompatibilities not already added by that.